### PR TITLE
[CARBONDATA-2954]Fix error when create external table command fired if path already exists

### PR DESCRIPTION
**Pull request by shardul-cr7**
Problem : Creating a external table and providing a valid location having some empty directory and .carbondata files was giving "operation not allowed: invalid datapath provided" error.

Solution: It was happening because if the location was having some empty directory getFilePathExternalFilePath method in carbonutil.java was returning null due to the presence of empty directory.So made a slight modification to prevent this problem.

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [x] Testing done
        manually tested.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/375/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8623/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/553/

**Comment by shardul-cr7** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/383/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8631/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/561/

File: core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java [hunk 1]
```diff
@@ -2226,7 +2226,11 @@ public static String getFilePathExternalFilePath(String path, Configuration conf
       if (dataFile.getName().endsWith(CarbonCommonConstants.FACT_FILE_EXT)) {
         return dataFile.getAbsolutePath();
       } else if (dataFile.isDirectory()) {
-        return getFilePathExternalFilePath(dataFile.getAbsolutePath(), configuration);
+        if (getFilePathExternalFilePath(dataFile.getAbsolutePath(), configuration) == null) {
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java]: Please  add/modify existing test case to handle this scenario

Also please test query with an empty folder. we have other places in the code where we recursively read.
Review and handle if a problem exists in that flow.

example:
CarbonTable.getFirstIndexFile --> also has same issue

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/427/

File: core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java (hunk 1 above)
Comment (shardul-cr7) [reply core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java]: Handled the review comment and added the test cases also for the scenarios.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/428/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/607/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8677/

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java [hunk 2]
```diff
@@ -261,7 +261,11 @@ private static CarbonFile getFirstIndexFile(CarbonFile tablePath) {
     CarbonFile[] carbonFiles = tablePath.listFiles();
     for (CarbonFile carbonFile : carbonFiles) {
       if (carbonFile.isDirectory()) {
-        return getFirstIndexFile(carbonFile);
+        if (getFirstIndexFile(carbonFile) == null) {
```
Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java]: Please add comments

File: core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java [hunk 3]
```diff
@@ -2230,7 +2230,11 @@ public static String getFilePathExternalFilePath(String path, Configuration conf
...
```
Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java]: Please add comments

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/432/

**Comment by kumarvishal09** [comment]
LGTM

**Comment by ajantha-bhat** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/612/
